x/concentrated-liquidity/swaps.go

```diff
@@ -47,13 +52,53 @@ func (k Keeper) CreateNewConcentratedLiquidityPool(ctx sdk.Context, poolId uint6
 func (k Keeper) SwapExactAmountIn(
 	ctx sdk.Context,
 	sender sdk.AccAddress,
-	pool swaproutertypes.PoolI,
+	poolI swaproutertypes.PoolI,
 	tokenIn sdk.Coin,
 	tokenOutDenom string,
 	tokenOutMinAmount sdk.Int,
 	swapFee sdk.Dec,
-) (sdk.Int, error) {
-	return sdk.Int{}, nil
+) (tokenOutAmount sdk.Int, err error) {
+	if tokenIn.Denom == tokenOutDenom {
+		return sdk.Int{}, errors.New("cannot trade same denomination in and out")
+	}
+
+	// get concentratedPool extension from poolI
+	pool, err := k.getPoolById(ctx, poolI.GetId())
+	if err != nil {
+		return sdk.Int{}, err
+	}
+
+	// determine if we are swapping asset0 for asset1 or vice versa
+	asset0 := pool.GetToken0()
+	zeroForOne := tokenIn.Denom == asset0
+	var tokenOutCoin sdk.Coin
+
+	// change priceLimit based on which direction we are swapping
+	if zeroForOne {
+		tokenOutCoin, err = k.SwapOutAmtGivenIn(ctx, tokenIn, tokenOutDenom, swapFee, sdk.NewDec(1), pool.GetId())
+	} else {
+		tokenOutCoin, err = k.SwapOutAmtGivenIn(ctx, tokenIn, tokenOutDenom, swapFee, sdk.NewDec(999999999999), pool.GetId())
+	}
```

Comment:
Let's define `sdk.NewDec(999999999999)` as `upperPriceLimit` and `sdk.OneDec()` as `lowerPriceLimit` 

```suggestion
        var priceLimit sdk.Dec
	// change priceLimit based on which direction we are swapping
	if zeroForOne {
		priceLimit = lowerPriceLimit
	} else {
		priceLimit = upperPriceLimit
	}
	tokenOutCoin, err = k.SwapOutAmtGivenIn(ctx, tokenIn, tokenOutDenom, swapFee, priceLimit, pool.GetId())
```